### ISSUE TITLE
Add Group role as supported required children element for ListBox role (1.2 specification update)

### ISSUE THREAD

**Issue by cscrosati** [dequelabs/axe-core#2523]
The WAI-ARIA 1.2 spec adds the `group` role with options children as a supported required owned elements of `listbox` ([refer to the draft spec reference](https://www.w3.org/TR/wai-aria-1.2/#listbox)).

We leverage this implementation in our platform since it is the recommended way to emulate a `<select>` with `<optgroup>` grouping. ([refer to the W3C ListBox with grouped options sample](https://w3c.github.io/aria-practices/examples/listbox/listbox-grouped.html))

While not completely thorough, an investigation of support of the 1.2 ListBox pattern is good among common browser and screen reader combos.

We think that axe-core should detect the `group` role with `option` as a valid required owned elements for `listbox`, as per the 1.2 specification updates to reflect current support and specification signal.

Currently, running axe-core (via Accessibility Insights for this issue report), there is errors reported for the [ListBox with grouped options sample](https://w3c.github.io/aria-practices/examples/listbox/listbox-grouped.html).

Expectation: No error reported.

Actual: Reports the following error:
Required ARIA child role not present: option

Motivation: The group role support is actually well supported in all browser when in a listbox role. While the readout isn't consistent across platform, it does show relationship and groupings. As such, allowing this role today, while technically in a draft spec, would reflect current support.

<pre><code>
axe-core version: 4.0.1
from Accessibility Insight version 2.21.1

Browser and Assistive Technology versions
Edge 85 + Narrator (excellent)
Chrome 85 + JAWS 2020 (excellent)
Firefox 80 + NVDA 2020.1 (great)
IE 11 + Narrator (ok)

</code></pre>

**Comment by WilcoFiers** [comment]
Hey @cscrosati. ARIA 1.2 isn't stable yet. I definitely wouldn't use this in production code, as this could all still change.

I took at look at the example. We'd need to test this with AT, but my guess is this pattern doesn't work consistently. Definitely one problem with it is that presentational group name. Putting role=presentation on an li doesn't make the text node go away. It throws off the set size count on the group. I'm a little surprised Chrome ignores it, probably because they too had realised this is how people were going to implement it, even though it creates issues. Firefox counts the content of the presentational li as children of the group though, as does Safari.

That's a problem with the way the ARIA practice recommends doing it, not necessarily using group in listbox. The right way to do it would be to use aria-hidden=true instead of role=presentation:

```html
...
<div id="ss_elem_list" tabindex="0" role="listbox" aria-labelledby="ss_elem">
<ul role="group" aria-labelledby="cat1">
  <li aria-hidden="true" id="cat1">Land</li>
  <li id="ss_elem_1" role="option">Cat</li>
  ...
```

Either way, we'll need to test support for this, but yes, if this works in everything we have in axe-core's [accessibility baseline](https://github.com/dequelabs/axe-core/blob/develop/doc/accessibility-supported.md), we'll add it to axe-core.

**Comment by driskull** [comment]
I'm seeing this issue as well with the aXe toolbar on [this page](https://www.w3.org/TR/wai-aria-practices-1.2/examples/listbox/listbox-grouped.html)

**Comment by dylanb** [comment]
Would be nice to have experimental variants of our rules

**Comment by smhigley** [comment]
I just tested the grouped listbox in the [APG page](https://w3c.github.io/aria-practices/examples/listbox/listbox-grouped.html) in all the browser/SR combinations listed in the [linked axe-core docs page](https://github.com/dequelabs/axe-core/blob/develop/doc/accessibility-supported.md).

I found all Windows browser/SRs and iOS/VO worked as expected (Firefox had no issues with indexing), and macOS VoiceOver and Talkback had indexing issues with both that example and the native select/optgroup.

Based on those results + there not being any better way of marking up grouped options, it seems appropriate to me to no longer flag this as a failure. The ARIA spec is also in CR now, and the chances of listbox > group becoming invalid before it goes into full rec are essentially zero.

Would axe-core be willing to update the rule? It's currently causing automated failures for Fluent, and multiple other products I work with.
